### DISCUSSION TITLE
Adding a toolbar

### DISCUSSION THREAD

**Discussion by DiggesT** [facebook/lexical#2649] [Q&A] [answered]
Dear Lexical team,

Can you please explain me, how to add toolbar in lexical?

I saw the toolbar as plugin on playground ([ToolbarPlugin.tsx](https://github.com/facebook/lexical/blob/5caf9231d456cadeb0f83fb69602b3bc6e433710/packages/lexical-playground/src/plugins/ToolbarPlugin.tsx)), but i can't find it in lexical.
Also, i cant' download this plugin.

So, do i have to write the toolbar manually?

**Comment by tomaszferens** [comment, marked as answer]
I was also very confused with this. From what I understand, you are responsible for writing some of the plugins, because they might be very tied to how your application works. Lexical gives you good building blocks for that. 

Playground is just to show what you can achieve with Lexical and acts as a good starting point for your own plugins.

**Reply by chuwy** [reply]
I _think_ that technically anyone can publish their own toolbar as a lexical plugin on NPM. Probably even copypasting code from playground. Just wondering why nobody has done it yet - the playground also seem to be MIT-licensed, so no legal restrictions.

**Reply by thegreatercurve** [reply]
@chuwy Someone has done that fairly recently: https://github.com/ozanyurtsever/verbum.

The reason we don't directly export the components from the playground, including the toolbar, is that we don't have the engineering resource to maintain the framework for building text editors AND a best-in-class example rich text editor.

The playground is a pretty simple example at the moment, but you can take what you like from there and use "as is" as it all falls under the same licence.

**Comment by FirstWhack** [comment]
For posterity, the toolbar plugin used in Lexical's playground as well as in Verbum are monolithic, untested, full of hardcoded states, and generally do not even function as intended (play with the color painter for example).

These should not even be referenced as examples.
